sw/device/lib/crypto/impl/kdf.c

```diff
@@ -28,8 +29,109 @@ otcrypto_status_t otcrypto_kdf_kmac(
     otcrypto_kmac_mode_t kmac_mode, const otcrypto_const_byte_buf_t kdf_label,
     const otcrypto_const_byte_buf_t kdf_context, size_t required_byte_len,
     otcrypto_blinded_key_t *keying_material) {
-  // TODO: Implement KMAC-KDF.
-  return OTCRYPTO_NOT_IMPLEMENTED;
+  // Check NULL pointers.
+  if (key_derivation_key.keyblob == NULL || keying_material == NULL) {
+    return OTCRYPTO_BAD_ARGS;
+  }
+
+  // Check the private key checksum.
+  if (integrity_blinded_key_check(&key_derivation_key) != kHardenedBoolTrue) {
+    return OTCRYPTO_BAD_ARGS;
+  }
+
+  // The underlying KMAC implementation is not hardened yet.
+  if (key_derivation_key.config.security_level !=
+      kOtcryptoKeySecurityLevelLow) {
+    return OTCRYPTO_NOT_IMPLEMENTED;
+  }
+
+  // Ensure `key_mode` is either KMAC-128 or KMAC-256.
+  if (key_derivation_key.config.key_mode != kOtcryptoKeyModeKdfKmac128 &&
+      key_derivation_key.config.key_mode != kOtcryptoKeyModeKdfKmac256) {
+    return OTCRYPTO_BAD_ARGS;
+  }
+
+  // Check if `key_mode` values match the input arg.
+  if (!((key_derivation_key.config.key_mode == kOtcryptoKeyModeKdfKmac128 &&
+         kmac_mode == kOtcryptoKmacModeKmac128) ||
+        (key_derivation_key.config.key_mode == kOtcryptoKeyModeKdfKmac256 &&
+         kmac_mode == kOtcryptoKmacModeKmac256))) {
+    return OTCRYPTO_BAD_ARGS;
+  }
```

Comment:
nit: I'd maybe replace these with checking the `key_derivation_key.config.key_mode` within the `if/else` block for `kmac_mode` below, so you can ensure it matches when you already know what `kmac_mode` is.